schemas/event.yml

```diff
@@ -744,3 +744,38 @@
         was taken. For example, a web proxy with an `event.action` which denied the request may also
         populate `event.reason` with the reason why (e.g. `blocked site`).
       example: "Terminated an unexpected process"
+
+    - name: agent_id_status
+      level: extended
+      type: keyword
+      short: Validation status of the event's agent.id field.
+      description: >
+        Agents are normally responsible for populating the `agent.id` field value.
+        If the system receiving events is capable of validating the value based
+        on authentication information for the client then this field can be used
+        to reflect the outcome of that validation.
+
+        For example if the agent's connection is authenticated with mTLS and
+        the client cert contains the ID of the agent to which the cert was issued
+        then the `agent.id` value in events can be checked against the certificate.
+        If the values match then `event.agent_id_status: verified` is added to
+        the event, otherwise one of the other allowed values should be used.
+
+        If no validation is performed then the field should be omitted.
+      example: verified
+      allowed_values:
```

Comment:
Setting `allowed_values` was introduced for the event category fields and, unfortunately, only works with those four fields right now. The attribute isn't very well documented right now, which should be fixed. It's also why we see the docs CI failure right now.

Can we move the allowed values into the `agent_id_status` description field and delete the `allowed_values` section?

I do like the idea of having a defined attribute to specify the allowed values outside of the event categorization fields, but implementing it is outside the scope here. 😄

Reply:
Done